lib/readline.js

```diff
@@ -47,9 +47,7 @@ function Interface(input, output, completer, terminal) {
   }
   historySize = historySize || kHistorySize;
 
-  completer = completer || function() { return []; };
-
-  if (typeof completer !== 'function') {
+  if (completer && typeof completer !== 'function') {
```

Comment:
That's my feeling. If you send it `false` then you are probably trying to say "Don't use a completer." In the spirit of being liberal with what is accepted but conservative with what is emitted, I wrote it up to allow falsy values. But strict checking for `undefined` would work too.